jan2025/client.js

```diff
@@ -0,0 +1,127 @@
+require("isomorphic-fetch");
+
+class Client {
+  constructor({ hasuraSecret, targetURL }) {
+    this.secret = hasuraSecret;
+    this.url = targetURL;
+  }
+
+  async graphQL(query, variables) {
+    const result = await fetch(this.url, {
+      method: "POST",
+      body: JSON.stringify({ query, variables }),
+      headers: {
+        "x-hasura-admin-secret": this.secret,
+      },
+    });
+    return result.json();
+  }
+
+  async getQueuedFlow() {
+    const { data, errors } = await this.graphQL(
+      `query GetQueuedFlow {
+        temp_data_migrations_audit(
+          limit: 1, 
+          where: {updated: {_eq: false}}
+        ) {
+          id: flow_id
+          flow {
+            slug
+            team {
+              slug
+            }
+            data
+            publishedFlows: published_flows(
+              limit: 1, 
+              order_by: {created_at: desc}
+            ) {
+              id  
+              data
+            }
+            sessions: lowcal_sessions(
+              where: {
+                created_at: {_gte: "2024-12-20"}, 
+                _or: {
+                  submitted_at: {_gte: "2024-12-20"}
```

Comment:
This is a key change based on over-fetching sessions during the `/dec2024` run and failing to correctly generate overview docs for recent submissions - instead of getting all records with null locked/submitted/deleted/sanitised timestamps, I'm instead fetching sessions:
- Created within last 28 days _or_
- Submitted within last 28 days

Together, these should ensure that both "active" sessions (can still be submitted) as well as recently submitted sessions still available to download via submissions log or email will both work as expected ("work" = validate against ODP Schema v0.7.2)

How does this seem? Anything I've potentially missed here?